crates/rpc/src/lib.rs

```diff
@@ -1,17 +1,86 @@
-use std::sync::Arc;
+use std::{collections::HashMap, sync::Arc, time::Duration};
 
 use actix_web::{App, HttpServer, dev::ServerHandle, middleware, web::Data};
 use config::RpcServerConfig;
+use ream_discv5::subnet::SyncCommitteeSubnets;
 use ream_p2p::network_state::NetworkState;
 use ream_storage::db::ReamDB;
+use tokio::sync::RwLock;
 use tracing::info;
 
-use crate::routes::register_routers;
+use crate::{handlers::validator::SyncCommitteeSubscriptionMap, routes::register_routers};
 
 pub mod config;
 pub mod handlers;
 pub mod routes;
 
+/// Spawns a background task to expire sync committee subscriptions.
+pub fn spawn_sync_committee_expiry_task(
+    sync_committee_subscriptions: SyncCommitteeSubscriptionMap,
+    sync_committee_subnets: Arc<RwLock<SyncCommitteeSubnets>>,
+    db: ReamDB,
+) {
+    tokio::spawn(async move {
+        loop {
+            // Fetch the current epoch from the latest state in the DB
+            let current_epoch = match get_current_epoch_from_db(&db).await {
+                Ok(epoch) => epoch,
+                Err(_) => {
+                    // If we can't get the epoch, skip this round
+                    tokio::time::sleep(Duration::from_secs(12 * 32)).await;
+                    continue;
+                }
+            };
+            let mut map = sync_committee_subscriptions.write().await;
+            let expired: Vec<u8> = map
+                .iter()
+                .filter_map(|(&subnet_id, &until_epoch)| {
+                    if until_epoch <= current_epoch {
+                        Some(subnet_id)
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            if !expired.is_empty() {
+                let mut subnets = sync_committee_subnets.write().await;
+                for subnet_id in &expired {
+                    if let Err(e) = subnets.disable_sync_committee_subnet(*subnet_id) {
+                        tracing::error!(
+                            "Failed to disable sync committee subnet {}: {}",
+                            subnet_id,
+                            e
+                        );
+                    }
+                    map.remove(subnet_id);
+                }
+
+                if !expired.is_empty() {
+                    tracing::info!(
+                        "Marked that ENR needs to be updated after sync committee subnet expiry"
+                    );
+                }
+            }
+            tokio::time::sleep(Duration::from_secs(12 * 32)).await; // One epoch (customize as needed)
+        }
+    });
+}
```

Comment:
can you make it so the expired subscriptions are handled in the NetworkManager loop { select! {} } maybe we can use a tokio interval to check every X seconds, set the Interval's behavior to skip